applications/ParticleMechanicsApplication/custom_utilities/mpm_search_element_utility.h

```diff
@@ -85,6 +88,42 @@ namespace MPMSearchElementUtility
                 // FindPointOnMesh find the background element in which a given point falls and the relative shape functions
                 bool is_found = SearchStructure.FindPointOnMesh(xg[0], N, pelem, result_begin, MaxNumberOfResults, Tolerance);
 
+
+                if (is_found && is_explicit) {
```

Comment:
hmmm
the `SearchElement` element function grew to 140 lines now...
I think something should be refactored in the future

Reply:
No worries, added in todo [here](https://github.com/KratosMultiphysics/Kratos/projects/17#card-36873805)